lib/generators/test_unit/webauthn_authentication/templates/test/controllers/second_factor_authentications_controller_test.rb

```diff
@@ -0,0 +1,95 @@
+require "test_helper"
+require "webauthn/fake_client"
+
+class SecondFactorAuthenticationsControllerTest < ActionDispatch::IntegrationTest
+  setup do
+    @user = users(:one)
+    @client = WebAuthn::FakeClient.new(WebAuthn.configuration.allowed_origins.first)
+
+    creation_options = WebAuthn::Credential.options_for_create(
+      user: { id: @user.webauthn_id, name: @user.email_address },
+      authenticator_selection: { resident_key: "discouraged", user_verification: "discouraged" }
+    )
+    create_options = @client.create(challenge: creation_options.challenge)
+    credential = WebAuthn::Credential.from_create(create_options)
+
+    WebauthnCredential.second_factor.create!(
+      nickname: "My Security Key",
+      user: @user,
+      external_id: credential.id,
+      public_key: credential.public_key,
+      sign_count: 0
+    )
+  end
+
+  test "should return get_options" do
+    post session_path, params: { email_address: @user.email_address, password: "password" }
+
+    post get_options_second_factor_authentication_url
+
+    assert_response :success
+    body = JSON.parse(response.body)
+    assert body["challenge"].present?
+    assert body["userVerification"] == "discouraged"
+
+    assert_equal session[:current_authentication][:challenge], body["challenge"]
+  end
+
+  test "should create session with valid credential" do
```

Comment:
What do you think about prepending the name of the action we are testing?


```suggestion
  test "#create should create session with valid credential" do
```

Reply:
Hmm I don't know if I like it tbh. Feels too verbose.
Is that a convention?

Reply:
The thing is that otherwise it's not clear at first glance which action we are actually testing 😕 

> I'll try to standardize all of them following this as a reference: https://github.com/rails/rails/blob/a88c419fc6033fb633ac24c084f54e37c38a5025/railties/lib/rails/generators/test_unit/authentication/templates/test/controllers/passwords_controller_test.rb.tt

Looks good to me!